### PR TITLE
Category module

### PR DESCRIPTION
**Pull request by hirengevariya**
Change all message as per the module / function response.
define findByTitle() module
error handle in addCategory() for insert data.

### CODE REVIEW

File: src/actions/categoryActions.js [hunk 1]
```diff
@@ -1,45 +1,51 @@
...
+            } else {
+                const {title} = req.body;
+                //Add Category
+                const categoryData = {title};
+                CategoryModel.insert(categoryData).then((createRes) => {
+                    res.json({status: true, data: createRes, message: 'New category added'});
+                })
+                    .catch(
+                        () => {
+                            res.json({status: false, message: 'Category Failed'})
```
Comment (suresh-jarvis) [inline src/actions/categoryActions.js]: `Category Failed` this is not correct message `Category create fail. Please try again.`
Do code format before commit
current code has 4 space tab we use 2 space tab

File: src/actions/categoryActions.js [hunk 2]
```diff
@@ -1,45 +1,51 @@
...
+                        }
+                    )
+                }
+          });
+    },
+
+    getByIdCategory: function (req, res) {
+        const { id } = req.params;
+        CategoryModel.findById(id).then((cidRes) => {
+            res.json({status: true, data:cidRes})
```
Comment (suresh-jarvis) [inline src/actions/categoryActions.js]: Use proper variable name not ony cidRes.

File: src/actions/categoryActions.js [hunk 3]
```diff
@@ -1,45 +1,51 @@
...
+        });
+    },
+
+   deleteCategory:function (req, res) {
+       const { id } = req.params;
+       CategoryModel.delete(id).then((data) => {
+           res.json({status: true, message:'Category deleted successfully.'})
```
Comment (suresh-jarvis) [inline src/actions/categoryActions.js]: It's Repeat : Use space after (:)

File: src/actions/categoryActions.js [hunk 4]
```diff
@@ -1,45 +1,51 @@
...
+       })
+    },
+
+    updateCategory:function (req, res) {
+        const { id } = req.params;
+        CategoryModel.update(id, req.body).then((cupRes) => {
+            res.json({status: true, message:'Category updated SuccessFully.', data:cupRes})
```
Comment (suresh-jarvis) [inline src/actions/categoryActions.js]: Message should have 1 word in uppercase and rest of words are in lowercase.

File: src/model/category.js [hunk 5]
```diff
@@ -2,77 +2,79 @@ const mongoose = require('mongoose');
...
-      new Category(categoryData).save(function (error, data) {
-        resolve((error) ? false : data)
-      })
-    })
-  },
+    insert: function (categoryData) {
+        categoryData['createdAt'] = moment();
+        return new Promise((resolve, reject) => {
+            new Category(categoryData).save(function (err, data) {
+                if (err == null) {
```
Comment (suresh-jarvis) [inline src/model/category.js]: Use ternory operator DOnt use if for single statement

**Comment by hirengevariya** [comment]
Resolve 2 Tab Space,
Change variable Name as per the function response,
Error handling for findByID() & deleteCategory() using ternary operator.

File: src/actions/categoryActions.js [hunk 6]
```diff
@@ -1,45 +1,55 @@
...
-  addCategory : function (req,res) {
-    const {title} = req.body;
-    const categoryData = {
-     title,
-      isVerified:true,
-    };
-    CategoryModel.insert(categoryData).then((categoryRes) => {
-      res.json({status: true, data:categoryRes, message:'Created SuccessFully!'})
+  getAllCategory: function (req, res) {
+    CategoryModel.getAll().then((callRes) => {
```
Comment (suresh-jarvis) [inline src/actions/categoryActions.js]: Response variable name should be realted to modal or table name like `categoriesRes`

File: src/actions/categoryActions.js [hunk 7]
```diff
@@ -1,45 +1,55 @@
...
 
-  getCategoryByTitle : function (req, res) {
-    const { title } = req.params;
-    CategoryModel.findByTitle(title).then((categoryRes) => {
-      res.json({status: true, data:categoryRes})
-    })
+  addCategory: function (req, res) {
+    CategoryModel.findByTitle(req.body.title).then((findTitleRes) => {
+      if (Object.keys(findTitleRes).length > 0) {
+        res.json({status: false, message: 'Category already exits'});
```
Comment (suresh-jarvis) [inline src/actions/categoryActions.js]: It should be `Category title already used. Please try with different title.`

File: src/actions/categoryActions.js [hunk 8]
```diff
@@ -1,45 +1,55 @@
...
+      } else {
+        const {title} = req.body;
+        //Add Category
+        const categoryData = {title};
```
Comment (suresh-jarvis) [inline src/actions/categoryActions.js]: No need to create variable for single ojbect key you can pass direct

File: src/actions/categoryActions.js [hunk 9]
```diff
@@ -1,45 +1,55 @@
...
+        CategoryModel.insert(categoryData).then((createRes) => {
+          console.log('New Cate Added', createRes);
```
Comment (suresh-jarvis) [inline src/actions/categoryActions.js]: Remove this console

File: src/actions/categoryActions.js [hunk 10]
```diff
@@ -1,45 +1,55 @@
...
 
-
-  getAllCategory :function (req,res) {
-    CategoryModel.getAll().then((categoryRes) => {
-      res.json({status: true, data: categoryRes})
+  getByIdCategory: function (req, res) {
+    const {id} = req.params;
+    CategoryModel.findById(id).then((categoryIdRes) => {
+      (!categoryIdRes)
+        ? res.json({status: true, message: 'Given data not match with exists database.'})
```
Comment (suresh-jarvis) [inline src/actions/categoryActions.js]: Message should be `Category not found!`

File: src/actions/categoryActions.js [hunk 11]
```diff
@@ -1,45 +1,55 @@
...
 
-  deleteCategoryById:function (req, res) {
-    const { id } = req.params;
-    CategoryModel.delete(id).then((response) => {
-      res.json({status: true, message:'Delete SuccessFully!!!.... '})
+  deleteCategory: function (req, res) {
+    const {id} = req.params;
+    CategoryModel.delete(id).then((deleteRes) => {
+      (!deleteRes)
+        ? res.json({status: true, message: 'Given data not match with exists database'})
```
Comment (suresh-jarvis) [inline src/actions/categoryActions.js]: Message : `Delete category fail! Please try again.`

File: src/actions/categoryActions.js [hunk 12]
```diff
@@ -1,45 +1,55 @@
...
 
-  updateCategoryById:function (req, res) {
-    const { id } = req.params;
-
-    CategoryModel.update(id, req.body).then((categoryRes) => {
-      res.json({status: true, message:'Update SuccessFully!!!.... ', data:categoryRes})
+  updateCategory: function (req, res) {
+    const {id} = req.params;
+    CategoryModel.update(id, req.body).then((categoryUpdateRes) => {
+      res.json({status: true, message: 'Category updated SuccessFully.', data: categoryUpdateRes})
```
Comment (suresh-jarvis) [inline src/actions/categoryActions.js]: Use `Successfully`. Replace with `SuccessFully`

File: src/model/category.js~ [hunk 13]
```diff
@@ -0,0 +1,67 @@
+const mongoose = require('mongoose');
```
Comment (suresh-jarvis) [inline src/model/category.js~]: You committed wrong file. `category.js~` it's system generated backup file. If you have edit with `gedit` linux notepad. So editor will generate that kind of flie.

Try to use `git status` in terminal before commit and check list of changes file which you want to commit.

**Comment by hirengevariya** [comment]
Resolve all previous comments.
Create a config file for the testing and know what is the use of config file.
while add new category to check whether upper / lower case to define schema toLowercase as a true by default, and then error handling
WIP - for Userid set into the category collection.

**Pull request merged**